### ISSUE TITLE
32 bit build fails on master: btl_base_am_rdma.c

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#8566]
32 bit builds are currently failing on master.  Git bisect says 3e317753f0308b917b4aa3b93b2d92997b9f491f is the bad commit:

```
make[1]: Entering directory `/home/jsquyres/git/o2/opal/mca/btl'
  CC       base/btl_base_am_rdma.lo
base/btl_base_am_rdma.c:199: warning: invalid use of structure with flexible array member
base/btl_base_am_rdma.c: In function ‘mca_btl_base_copy_from_segments’:
base/btl_base_am_rdma.c:294: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c:302: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c: In function ‘mca_btl_base_copy_to_segments’:
base/btl_base_am_rdma.c:324: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c:336: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_rdma_advance’:
base/btl_base_am_rdma.c:386: warning: cast from pointer to integer of different size
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_descriptor_complete’:
base/btl_base_am_rdma.c:415: warning: unused variable ‘context’
base/btl_base_am_rdma.c: In function ‘mca_btl_base_rdma_retry_operation’:
base/btl_base_am_rdma.c:724: warning: unused variable ‘btl’
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_rdma_progress’:
base/btl_base_am_rdma.c:770: warning: ISO C forbids braced-groups within expressions
base/btl_base_am_rdma.c:770: warning: ISO C forbids braced-groups within expressions
base/btl_base_am_rdma.c:779: warning: ISO C forbids braced-groups within expressions
base/btl_base_am_rdma.c:779: warning: ISO C forbids braced-groups within expressions
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_atomic_64’:
base/btl_base_am_rdma.c:804: error: implicit declaration of function ‘opal_atomic_fetch_add_64’
base/btl_base_am_rdma.c:807: error: implicit declaration of function ‘opal_atomic_fetch_and_64’
base/btl_base_am_rdma.c:810: error: implicit declaration of function ‘opal_atomic_fetch_or_64’
base/btl_base_am_rdma.c:813: error: implicit declaration of function ‘opal_atomic_fetch_xor_64’
base/btl_base_am_rdma.c:816: error: implicit declaration of function ‘opal_atomic_swap_64’
base/btl_base_am_rdma.c:819: error: implicit declaration of function ‘opal_atomic_fetch_min_64’
base/btl_base_am_rdma.c:822: error: implicit declaration of function ‘opal_atomic_fetch_max_64’
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_rdma_response’:
base/btl_base_am_rdma.c:895: error: implicit declaration of function ‘opal_atomic_add_fetch_64’
base/btl_base_am_rdma.c:894: warning: comparison between signed and unsigned integer expressions
base/btl_base_am_rdma.c: In function ‘mca_btl_base_am_process_atomic’:
base/btl_base_am_rdma.c:973: warning: pointer targets in passing argument 1 of ‘mca_btl_base_am_atomic_32’ differ in signedness
base/btl_base_am_rdma.c:832: note: expected ‘int32_t *’ but argument is of type ‘uint32_t *’
base/btl_base_am_rdma.c:977: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c:985: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c:990: error: implicit declaration of function ‘opal_atomic_compare_exchange_strong_64’
base/btl_base_am_rdma.c:990: warning: cast to pointer from integer of different size
base/btl_base_am_rdma.c:955: warning: unused variable ‘target_address’
base/btl_base_am_rdma.c: At top level:
base/btl_base_am_rdma.c:1008: warning: no previous prototype for ‘mca_btl_sm_sc_emu_init’
make[1]: *** [base/btl_base_am_rdma.lo] Error 1
make[1]: Leaving directory `/home/jsquyres/git/o2/opal/mca/btl'
make: *** [all-recursive] Error 1
```

A persistent question: do we care about 32 bit builds any more?

I think we have not formally removed 32 bit support, otherwise Past Jeff would likely have put in a configure check to disallow 32 bit builds.

**Comment by hjelmn** [comment]
I can fix but I at this point way are we supporting 32-bit? Does anyone care about i386, PPC32/Power32, ARM v6 and earlier, etc?

**Comment by rhc54** [comment]
We are going to discuss it tomorrow - don't do anything until then. We'll post the conclusion here. Right now, it isn't clear we care about 32-bit any more.

**Comment by jsquyres** [comment]
Thanks for the clarification, @rhc54 -- I didn't expect that @hjelmn would have noticed this issue so quickly.  :stuck_out_tongue_closed_eyes:

**Comment by hjelmn** [comment]
I was aware my code would not build 32-bit but I thought we had long-since axed that. It certainly is not tested by CI. I **may** call in tomorrow if I find time.

**Comment by gpaulsen** [comment]
Can we now close this?  for master/v5.0x?

**Comment by awlauria** [comment]
Master PR: https://github.com/open-mpi/ompi/pull/8722

Reopening since has not been merged to v5.0.x yet

**Comment by awlauria** [comment]
Closing as it was merged to v5.0.x here: https://github.com/open-mpi/ompi/pull/8902
